### PR TITLE
Add data structures to represent a dataset

### LINKED ISSUE #9: Create data model
## Description

The required data to calculate tau will be read in the ingest module, to facilitate idiomatic handling of the data in other modules and abstract away formats of persistent storage an internal representation needs to be created. This representation should be based on data structures which precisely encode the required data as precisely as possible.

The data structures should be broad enough to handle all realistic inputs, while disallowing any corrupted input/usage. This will play into the long term goal of making the system maintainable by encoding assumptions directly in the backbone of the system.

##  Acceptance Requirements

- All data required to calculate tau can be represented with the provided data structures
- The data structures reflect all constraints/assumptions of the system

### PR DESCRIPTION
**Pull request by Benedikt-Brunner**
### Summary

This adds all relevant data structures to represent the dataset of an experiment determining nuclear lifetimes.

There are also some format fixes across the repository.

---

### PR checklist

- [x] I have written tests that fail without my changes. _Or I did not add them on purpose._
- [x] I have included instructions on how to test this, assuming they are not already apparent from the linked issue.

---

### Depends on

- [x] Depends on #11 

### Closes/Fixes/Related to

* Closes #9

### CODE REVIEW

File: napytau/ingest/model/datapoint_collection.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+from typing import Dict, List, Callable, Iterator
+
+from napytau.ingest.model.datapoint import Datapoint
+from napytau.util.coalesce import coalesce
+from napytau.util.model.value_error_pair import ValueErrorPair
+
+
+class DatapointCollection:
```
Comment (julius-kress) [inline napytau/ingest/model/datapoint_collection.py:9]: I know it's an easy to understand data class, but perhaps a short comment would also be useful for the documentation.

File: napytau/ingest/model/dataset.py
Comment (julius-kress) [inline napytau/ingest/model/dataset.py:1]: The data class could probably be made even clearer using Python's "dataclass" annotation:
~~~
@dataclass
class DataSet:
    relative_velocity: RelativeVelocity
    datapoints: DatapointCollection
~~~

**Review by julius-kress — COMMENTED** [review]
I really like the code and the tests, all is written very clean 👍
Maybe some additional comments on some data classes, even if they are already good to understand.

File: napytau/ingest/model/dataset.py
Comment (Benedikt-Brunner) [reply napytau/ingest/model/dataset.py:1]: Thats a cool feature, didn't know that exists. I will add it to all trivial data classes 👍

**Review by julius-kress — APPROVED** [review]
Very nice, good job 🤗

**Pull request merged**